# Retrieve data from schema

In [3]:
import os
from dotenv import load_dotenv
import pandas as pd

load_dotenv()

DATABASE = os.getenv("DATABASE")
USER_DB = os.getenv("USER_DB")
PASSWORD = os.getenv("PASSWORD")
HOST = os.getenv("HOST")
PORT = os.getenv("PORT")

In [4]:
from sqlalchemy import create_engine

# read the database string from the .env
load_dotenv()

DB_STRING = os.getenv("DB_STRING")

if DB_STRING is None:
    raise ValueError("DB_STRING is not set in the environment.")

db = create_engine(DB_STRING)

In [43]:
query_string = ('SELECT DISTINCT ON (s.house_id) s.date, s.price, d.* FROM eda.king_county_house_sales AS s '
 'JOIN eda.king_county_house_details AS d ON d.id = s.house_id '
 'ORDER BY s.house_id, s.date DESC;')
df_eda = pd.read_sql(query_string, db)

In [44]:
df_eda.head(10)

,date,price,id,bedrooms,bathrooms,sqft_living,sqft_lot,floors,waterfront,view,...,grade,sqft_above,sqft_basement,yr_built,yr_renovated,zipcode,lat,long,sqft_living15,sqft_lot15
0,2015-04-22,300000.0,1000102,6.0,3.00,2400.0,9373.0,2.0,NaN,0.0,...,7,2400.0,0.0,1991,0.0,98002,47.3262,-122.214,2060.0,7316.0
1,2014-05-08,647500.0,1200019,4.0,1.75,2060.0,26036.0,1.0,NaN,0.0,...,8,1160.0,900.0,1947,0.0,98166,47.4444,-122.351,2590.0,21891.0
2,2014-08-11,400000.0,1200021,3.0,1.00,1460.0,43000.0,1.0,0.0,0.0,...,7,1460.0,0.0,1952,0.0,98166,47.4434,-122.347,2250.0,20023.0
3,2015-04-01,235000.0,2800031,3.0,1.00,1430.0,7599.0,1.5,0.0,0.0,...,6,1010.0,420.0,1930,0.0,98168,47.4783,-122.265,1290.0,10320.0
4,2015-03-19,402500.0,3600057,4.0,2.00,1650.0,3504.0,1.0,0.0,0.0,...,7,760.0,890.0,1951,20130.0,98144,47.5803,-122.294,1480.0,3504.0
5,2015-03-30,680000.0,3600072,4.0,2.75,2220.0,5310.0,1.0,NaN,0.0,...,7,1170.0,1050.0,1951,NaN,98144,47.5801,-122.294,1540.0,4200.0
6,2015-02-24,178000.0,3800008,5.0,1.50,1990.0,18200.0,1.0,NaN,0.0,...,7,1990.0,0.0,1960,NaN,98178,47.4938,-122.262,1860.0,8658.0
7,2014-07-09,487000.0,5200087,4.0,2.50,2540.0,5001.0,2.0,0.0,0.0,...,9,2540.0,0.0,2005,0.0,98108,47.5423,-122.302,2360.0,6834.0
8,2014-11-12,281000.0,6200017,3.0,1.00,1340.0,21336.0,1.5,0.0,0.0,...,5,1340.0,0.0,1945,0.0,98032,47.4023,-122.273,1340.0,37703.0
9,2014-11-04,239000.0,7200080,4.0,2.00,1980.0,10585.0,1.5,0.0,0.0,...,6,1980.0,0.0,1924,0.0,98055,47.4836,-122.214,1360.0,7810.0


Because we don't want to run the queries over and over again we can export the data into a .csv file in order to use it in other notebooks as well. 

In [45]:
# export the data to a csv-file
df_eda.to_csv("data/eda.csv", index=False)

In [46]:
# import the data from a csv-file
df_import = pd.read_csv("data/eda.csv")

In [47]:
df_import.head(10)

,date,price,id,bedrooms,bathrooms,sqft_living,sqft_lot,floors,waterfront,view,...,grade,sqft_above,sqft_basement,yr_built,yr_renovated,zipcode,lat,long,sqft_living15,sqft_lot15
0,2015-04-22,300000.0,1000102,6.0,3.00,2400.0,9373.0,2.0,NaN,0.0,...,7,2400.0,0.0,1991,0.0,98002,47.3262,-122.214,2060.0,7316.0
1,2014-05-08,647500.0,1200019,4.0,1.75,2060.0,26036.0,1.0,NaN,0.0,...,8,1160.0,900.0,1947,0.0,98166,47.4444,-122.351,2590.0,21891.0
2,2014-08-11,400000.0,1200021,3.0,1.00,1460.0,43000.0,1.0,0.0,0.0,...,7,1460.0,0.0,1952,0.0,98166,47.4434,-122.347,2250.0,20023.0
3,2015-04-01,235000.0,2800031,3.0,1.00,1430.0,7599.0,1.5,0.0,0.0,...,6,1010.0,420.0,1930,0.0,98168,47.4783,-122.265,1290.0,10320.0
4,2015-03-19,402500.0,3600057,4.0,2.00,1650.0,3504.0,1.0,0.0,0.0,...,7,760.0,890.0,1951,20130.0,98144,47.5803,-122.294,1480.0,3504.0
5,2015-03-30,680000.0,3600072,4.0,2.75,2220.0,5310.0,1.0,NaN,0.0,...,7,1170.0,1050.0,1951,NaN,98144,47.5801,-122.294,1540.0,4200.0
6,2015-02-24,178000.0,3800008,5.0,1.50,1990.0,18200.0,1.0,NaN,0.0,...,7,1990.0,0.0,1960,NaN,98178,47.4938,-122.262,1860.0,8658.0
7,2014-07-09,487000.0,5200087,4.0,2.50,2540.0,5001.0,2.0,0.0,0.0,...,9,2540.0,0.0,2005,0.0,98108,47.5423,-122.302,2360.0,6834.0
8,2014-11-12,281000.0,6200017,3.0,1.00,1340.0,21336.0,1.5,0.0,0.0,...,5,1340.0,0.0,1945,0.0,98032,47.4023,-122.273,1340.0,37703.0
9,2014-11-04,239000.0,7200080,4.0,2.00,1980.0,10585.0,1.5,0.0,0.0,...,6,1980.0,0.0,1924,0.0,98055,47.4836,-122.214,1360.0,7810.0


In [48]:
df_import.describe()

,price,id,bedrooms,bathrooms,sqft_living,sqft_lot,floors,waterfront,view,condition,grade,sqft_above,sqft_basement,yr_built,yr_renovated,zipcode,lat,long,sqft_living15,sqft_lot15
count,2.142000e+04,2.142000e+04,21420.000000,21420.000000,21420.000000,2.142000e+04,21420.000000,19060.000000,21357.000000,21420.000000,21420.000000,21420.000000,20969.000000,21420.000000,17609.000000,21420.00000,21420.000000,21420.000000,21420.000000,21420.000000
mean,5.418614e+05,4.580940e+09,3.373950,2.118429,2083.132633,1.512804e+04,1.495985,0.007660,0.234677,3.410784,7.662792,1791.170215,292.086938,1971.092997,838.805724,98077.87437,47.560197,-122.213784,1988.384080,12775.718161
std,3.675569e+05,2.876761e+09,0.925405,0.768720,918.808412,4.153080e+04,0.540081,0.087188,0.766459,0.650035,1.171971,828.692965,442.695597,29.387141,4005.127226,53.47748,0.138589,0.140791,685.537057,27345.621867
min,7.800000e+04,1.000102e+06,1.000000,0.500000,370.000000,5.200000e+02,1.000000,0.000000,0.000000,1.000000,3.000000,370.000000,0.000000,1900.000000,0.000000,98001.00000,47.155900,-122.519000,399.000000,651.000000
25%,3.249500e+05,2.123537e+09,3.000000,1.750000,1430.000000,5.040000e+03,1.000000,0.000000,0.000000,3.000000,7.000000,1200.000000,0.000000,1952.000000,0.000000,98033.00000,47.471200,-122.328000,1490.000000,5100.000000
50%,4.505500e+05,3.904921e+09,3.000000,2.250000,1920.000000,7.614000e+03,1.500000,0.000000,0.000000,3.000000,7.000000,1560.000000,0.000000,1975.000000,0.000000,98065.00000,47.572100,-122.230000,1840.000000,7620.000000
75%,6.450000e+05,7.308900e+09,4.000000,2.500000,2550.000000,1.069050e+04,2.000000,0.000000,0.000000,4.000000,8.000000,2220.000000,560.000000,1997.000000,0.000000,98117.00000,47.678100,-122.125000,2370.000000,10086.250000
max,7.700000e+06,9.900000e+09,33.000000,8.000000,13540.000000,1.651359e+06,3.500000,1.000000,4.000000,5.000000,13.000000,9410.000000,4820.000000,2015.000000,20150.000000,98199.00000,47.777600,-121.315000,6210.000000,871200.000000


# Stats
1. There are 21420 houses for sale
2. In the following columns there are NaN's: waterfront, view, sqft_basement, yr_renovated
3. They can be filled by zeros